### PR TITLE
Separate linting and tests from building and pushing docker image

### PR DESCRIPTION
**Pull request by n3h3m**
Two workflows for Alchemist
1 - To just linting and running tests upon every commit on any branch
2 - Build, tag, and push container image on every merging of pull request to `main` branch.

### CODE REVIEW

File: .github/workflows/linting_and_tests.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Upon every commit to any branch, do the linting, run tests
+
+on:
+  push:
+    paths:
+      - '**'
+env:
+  IMAGE_NAME: opendatacube/datacube-alchemist
```
Comment (alexgleith) [inline .github/workflows/linting_and_tests.yml:8]: We don't need the IMAGE_NAME

File: .github/workflows/linting_and_tests.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      with:
+        fetch-depth: 0
+
+    - name: Build the docker container
+      run: |
+        docker-compose build
+
+    - name: Bring up the docker container
+      run: |
+        docker-compose up -d
```
Comment (alexgleith) [inline .github/workflows/linting_and_tests.yml:25]: I don't think there's actually value in having these separate.

File: .github/workflows/linting_and_tests.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    - name: Run linting
+      run: |
+        docker-compose exec -T alchemist flake8
+
+    - name: Run tests
+      run: |
+        docker-compose exec -T alchemist pytest tests
```
Comment (alexgleith) [inline .github/workflows/linting_and_tests.yml:33]: No make commands any more?

File: .github/workflows/build_tag_push_image.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+name: When a pull request is merged to main, build the container, tag and push to the docker hub.
+
+on:
+  pull_request:
```
Comment (alexgleith) [inline .github/workflows/build_tag_push_image.yml:4]: Do you think this should happen on push to the main branch too? We haven't blocked that yet, I don't think...

**Review by alexgleith — CHANGES_REQUESTED** [review]
Few little changes suggested

File: .github/workflows/linting_and_tests.yml (hunk 3 above)
Comment (n3h3m) [reply .github/workflows/linting_and_tests.yml:33]: Easy to know what it is doing, If it's from Makefile, then some has to click the Makefile to see what command is invoked.

File: .github/workflows/build_tag_push_image.yml (hunk 4 above)
Comment (n3h3m) [reply .github/workflows/build_tag_push_image.yml:4]: I can add it, but better to block direct commits to `main`.

File: .github/workflows/build_tag_push_image.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+jobs:
+  push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: main
```
Comment (alexgleith) [inline .github/workflows/build_tag_push_image.yml:26]: Do we need to specify the branch here?

**Pull request closed without merging**